### PR TITLE
New cross region replication mm2 converter

### PR DESCRIPTION
**Pull request by JC-CJ2222**
Created MM2 Converter, Added Unit Tests, Modified based on First CR Feedback

### CODE REVIEW

File: common/pom.xml [hunk 1]
```diff
@@ -65,6 +65,14 @@
             <groupId>software.amazon.awssdk</groupId>
             <artifactId>glue</artifactId>
         </dependency>
+
+        <dependency>
+            <groupId>org.apache.maven.plugins</groupId>
```
Comment (YangSan0622) [inline common/pom.xml]: Why do we need this dependency?

File: common/src/main/java/com/amazonaws/services/schemaregistry/common/AWSSchemaRegistryClient.java [hunk 2]
```diff
@@ -90,18 +90,19 @@ public AWSSchemaRegistryClient(@NonNull AwsCredentialsProvider credentialsProvid
                 .retryPolicy(retryPolicy)
                 .addExecutionInterceptor(new UserAgentRequestInterceptor())
                 .build();
-        UrlConnectionHttpClient.Builder urlConnectionHttpClientBuilder = UrlConnectionHttpClient.builder();
-        if (glueSchemaRegistryConfiguration.getProxyUrl() != null) {
-        	log.debug("Creating http client using proxy {}", glueSchemaRegistryConfiguration.getProxyUrl().toString());
-    		ProxyConfiguration proxy = ProxyConfiguration.builder().endpoint(glueSchemaRegistryConfiguration.getProxyUrl()).build();
-    		urlConnectionHttpClientBuilder.proxyConfiguration(proxy);
-        }
+//        UrlConnectionHttpClient.Builder urlConnectionHttpClientBuilder = UrlConnectionHttpClient.builder();
```
Comment (YangSan0622) [inline common/src/main/java/com/amazonaws/services/schemaregistry/common/AWSSchemaRegistryClient.java]: Please do not include commented out code in Pull Request. Delete them if they should not be included.

Comment (YangSan0622) [reply common/src/main/java/com/amazonaws/services/schemaregistry/common/AWSSchemaRegistryClient.java]: Why do we need to remove our http client builder in this case?

File: common/src/main/java/com/amazonaws/services/schemaregistry/common/configs/GlueSchemaRegistryConfiguration.java [hunk 3]
```diff
@@ -86,6 +88,10 @@ private void buildConfigs(Map<String, ?> configs) {
     }
 
     private void buildSchemaRegistryConfigs(Map<String, ?> configs) {
+        validateAndSetAWSSourceRegion(configs);
+        validateAndSetAWSSourceEndpoint(configs);
```
Comment (YangSan0622) [inline common/src/main/java/com/amazonaws/services/schemaregistry/common/configs/GlueSchemaRegistryConfiguration.java]: the formatting looks a little bit strange here, can you format the code?

File: common/src/main/java/com/amazonaws/services/schemaregistry/common/configs/GlueSchemaRegistryConfiguration.java [hunk 4]
```diff
@@ -101,7 +107,6 @@ private void buildSchemaRegistryConfigs(Map<String, ?> configs) {
         validateAndSetMetadata(configs);
         validateAndSetUserAgent(configs);
         validateAndSetSecondaryDeserializer(configs);
-        validateAndSetProxyUrl(configs);
```
Comment (YangSan0622) [inline common/src/main/java/com/amazonaws/services/schemaregistry/common/configs/GlueSchemaRegistryConfiguration.java]: why this is removed?

File: common/src/main/java/com/amazonaws/services/schemaregistry/common/configs/GlueSchemaRegistryConfiguration.java [hunk 5]
```diff
@@ -140,7 +145,7 @@ private boolean validateCompressionType(String compressionType) {
         if (!EnumUtils.isValidEnum(AWSSchemaRegistryConstants.COMPRESSION.class, compressionType.toUpperCase())) {
             String errorMessage =
                     String.format("Invalid Compression type : %s, Accepted values are : %s", compressionType,
-                                  AWSSchemaRegistryConstants.COMPRESSION.values());
+                            AWSSchemaRegistryConstants.COMPRESSION.values());
```
Comment (YangSan0622) [inline common/src/main/java/com/amazonaws/services/schemaregistry/common/configs/GlueSchemaRegistryConfiguration.java]: redundant change, please remove it

File: common/src/main/java/com/amazonaws/services/schemaregistry/common/configs/GlueSchemaRegistryConfiguration.java [hunk 6]
```diff
@@ -154,17 +159,31 @@ private void validateAndSetAWSRegion(Map<String, ?> configs) {
...
+
     private void validateAndSetCompatibility(Map<String, ?> configs) {
         if (isPresent(configs, AWSSchemaRegistryConstants.COMPATIBILITY_SETTING)) {
             this.compatibilitySetting = Compatibility.fromValue(
                     String.valueOf(configs.get(AWSSchemaRegistryConstants.COMPATIBILITY_SETTING))
                             .toUpperCase());
 
             if (this.compatibilitySetting == null
-                || this.compatibilitySetting == Compatibility.UNKNOWN_TO_SDK_VERSION) {
+                    || this.compatibilitySetting == Compatibility.UNKNOWN_TO_SDK_VERSION) {
```
Comment (YangSan0622) [inline common/src/main/java/com/amazonaws/services/schemaregistry/common/configs/GlueSchemaRegistryConfiguration.java]: redundant change, please remove this one and other similar changes

File: common/src/main/java/com/amazonaws/services/schemaregistry/common/configs/GlueSchemaRegistryConfiguration.java [hunk 7]
```diff
@@ -186,18 +205,21 @@ private void validateAndSetAWSEndpoint(Map<String, ?> configs) {
...
+
+    private void validateAndSetAWSTargetEndpoint(Map<String, ?> configs) {
+        if (isPresent(configs, AWSSchemaRegistryConstants.AWS_TARGET_ENDPOINT)) {
+            this.targetEndPoint = String.valueOf(configs.get(AWSSchemaRegistryConstants.AWS_TARGET_ENDPOINT));
+        } else {
+            this.targetEndPoint = String.valueOf(configs.get(AWSSchemaRegistryConstants.AWS_ENDPOINT));
         }
     }
 
+
```
Comment (YangSan0622) [inline common/src/main/java/com/amazonaws/services/schemaregistry/common/configs/GlueSchemaRegistryConfiguration.java:236]: redundant line break

File: common/src/main/java/com/amazonaws/services/schemaregistry/utils/AWSSchemaRegistryConstants.java [hunk 8]
```diff
@@ -18,10 +18,6 @@
 import software.amazon.awssdk.services.glue.model.Compatibility;
 
 public final class AWSSchemaRegistryConstants {
-    /**
-     * Proxy URL to use while connecting to AWS endpoint.
-     */
-    public static final String PROXY_URL = "proxyUrl";
```
Comment (YangSan0622) [inline common/src/main/java/com/amazonaws/services/schemaregistry/utils/AWSSchemaRegistryConstants.java]: why proxy url is removed?

File: cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java [hunk 9]
```diff
@@ -0,0 +1,157 @@
...
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+public class CrossRegionReplicationMM2Converter implements Converter {
+    @Getter
+    private GlueSchemaRegistryDeserializationFacade deserializationFacade;
```
Comment (YangSan0622) [inline cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java:30]: Why do we need Getter annotation for the private fields of the Converter class?

File: cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java [hunk 10]
```diff
@@ -0,0 +1,157 @@
...
+    private GlueSchemaRegistryDeserializerImpl deserializer;
+    @Getter
+    private GlueSchemaRegistrySerializerImpl serializer;
+    @Getter
+    private boolean isKey;
+
+    /**
+     * Constructor used by Kafka Connect user.
+     */
+    public CrossRegionReplicationMM2Converter(){};
```
Comment (YangSan0622) [inline cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java:40]: Is this constructor used by Connector? if it is the case, how will the fields be initialized?

File: cross-region-replication-mm2-converter/src/main/resources/archetype-resources/src/main/java/App.java [hunk 11]
```diff
@@ -0,0 +1,13 @@
+package $org.example;
```
Comment (YangSan0622) [inline cross-region-replication-mm2-converter/src/main/resources/archetype-resources/src/main/java/App.java]: what is the purpose of this module?

File: cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java [hunk 12]
```diff
@@ -0,0 +1,157 @@
...
+
+    /**
+     * Constructor accepting AWSCredentialsProvider.
+     *
+     * @param credentialsProvider AWSCredentialsProvider instance.
+     */
+    public CrossRegionReplicationMM2Converter(
```
Comment (YangSan0622) [inline cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java:47]: When will this constructor be used?

File: cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java [hunk 13]
```diff
@@ -0,0 +1,157 @@
...
+        new CrossRegionReplicationMM2ConverterConfig(configs);
+
+
+        credentialsProvider = DefaultCredentialsProvider.builder().build();
+
+        // Put the source and target regions into configurations respectively
+        Map<String, Object> sourceConfigs = new HashMap<>(configs);
+        Map<String, Object> targetConfigs = new HashMap<>(configs);
+
+        sourceConfigs.put(AWSSchemaRegistryConstants.AWS_REGION, configs.get(AWSSchemaRegistryConstants.AWS_SOURCE_REGION));
```
Comment (YangSan0622) [inline cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java:86]: Have we handle the case where expected key-value pair does not exist in the config map?

**Comment by YangSan0622** [comment]
Can you modify the target branch of this pull request to https://github.com/awslabs/aws-glue-schema-registry/tree/schema-replication-converter?

File: cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java (hunk 10 above)
Comment (JC-CJ2222) [reply cross-region-replication-mm2-converter/src/main/java/com/amazonaws/services/crossregion/schemaregistry/kafkaconnect/CrossRegionReplicationMM2Converter.java:40]: Both constructors here are used for testing purpose

**Pull request closed without merging**